src/modules/time.rs

```diff
@@ -0,0 +1,30 @@
+use ansi_term::Color;
+use chrono::{DateTime, Local};
+
+use super::{Context, Module};
+
+/// Outputs the current time
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("time")?;
+
+    let mut time_format = module
+        .config_value_str("format")
+        .unwrap_or("%H:%M")
+        .to_owned();
+
+    let is_12hr = module.config_value_bool("12hr").unwrap_or(false);
+    if is_12hr {
+        time_format = module
+            .config_value_str("format")
+            .unwrap_or("%I:%M %p")
+            .to_owned();
+    }
```

Comment:
Personally, I would style the above block as 

```rust
let is_12hr = module.config_value_bool("12hr").unwrap_or(false);
let default_format = if is_12hr { "%I:%M:%p" } else { "%H:%M" };
let time_format = module. 
            .config_value_str("format")
            .unwrap_or(default_format)
            .to_owned();
```
(Or something like that, I haven't actually tested yet).

Not a huge deal, but it does avoid repeating the lookup into the module.

Reply:
Thank you so much @chipbuster!